pennylane/ops/op_math/linear_combination.py

```diff
@@ -271,27 +272,37 @@ def compute_grouping(self, grouping_type="qwc", method="rlf"):
 
         _, ops = self.terms()
 
-        with qml.QueuingManager.stop_recording():
-            op_groups = qml.pauli.group_observables(ops, grouping_type=grouping_type, method=method)
-
-        ops = copy(ops)
-
-        indices = []
-        available_indices = list(range(len(ops)))
-        for partition in op_groups:  # pylint:disable=too-many-nested-blocks
-            indices_this_group = []
-            for pauli_word in partition:
-                # find index of this pauli word in remaining original observables,
-                for ind, observable in enumerate(ops):
-                    if qml.pauli.are_identical_pauli_words(pauli_word, observable):
-                        indices_this_group.append(available_indices[ind])
-                        # delete this observable and its index, so it cannot be found again
-                        ops.pop(ind)
-                        available_indices.pop(ind)
-                        break
-            indices.append(tuple(indices_this_group))
-
-        self._grouping_indices = tuple(indices)
+        # 'rlf' method is not compatible with the rx implementation.
+        if method == "rlf":
+            with qml.QueuingManager.stop_recording():
+                op_groups = qml.pauli.group_observables(
+                    ops, grouping_type=grouping_type, method=method
+                )
+
+            ops = copy(ops)
+
+            indices = []
+            available_indices = list(range(len(ops)))
+            for partition in op_groups:  # pylint:disable=too-many-nested-blocks
+                indices_this_group = []
+                for pauli_word in partition:
+                    # find index of this pauli word in remaining original observables,
+                    for ind, observable in enumerate(ops):
+                        if qml.pauli.are_identical_pauli_words(pauli_word, observable):
+                            indices_this_group.append(available_indices[ind])
+                            # delete this observable and its index, so it cannot be found again
+                            ops.pop(ind)
+                            available_indices.pop(ind)
+                            break
+                indices.append(tuple(indices_this_group))
+
+                grouping_indices = tuple(indices)
+        else:
```

Comment:
Might read easier if we switch the order of execution in this if statement.

```
if method != "rlf":
    self._grouping_indiices = qml.pauli.compute_partition_indices(ops, grouping_type=grouping_type, method=method)
    return 
# then rlf logic here
```

Potentially, we could also move the above logic into `compute_partition_indices`.

Reply:
Great suggestion, I am adding this to the compute_partition_indices logic!